File: Easy/3099_harshad-number/Solution.py
class Solution:
    def sumOfTheDigitsOfHarshadNumber(self, x: int) -> int:
        original = x
        total_sum = 0

        while x > 0:
            total_sum += x % 10
            x //= 10

        if original % total_sum == 0:
            return total_sum

        return -1

File: Easy/3099_harshad-number/Solution.js
class Solution {
    /**
     * @param {number} x
     * @return {number}
     */
    sumOfTheDigitsOfHarshadNumber(x) {
        let original = x;
        let sum = 0;

        while (x > 0) {
            sum += x % 10;
            x = Math.floor(x / 10);
        }

        if (original % sum === 0) {
            return sum;
        }

        return -1;
    }
}